datumaro/plugins/data_formats/arrow/arrow_dataset.py

```diff
@@ -0,0 +1,203 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import copy
+import errno
+import os
+from collections.abc import Iterator
+from typing import Dict, List, Optional, Tuple, TypeVar, Union, overload
+
+import numpy as np
+import pyarrow as pa
+
+PathLike = TypeVar("PathLike", str, os.PathLike, None)
+
+
+def _binary_search(arr, x, low=None, high=None, fn=None):
+    low = 0 if low is None else low
+    high = len(arr) - 1 if high is None else high
+    if fn is None:
+
+        def fn(arr, x):
+            return arr[x]
+
+    while low <= high:
+        mid = (high + low) // 2
+        val = fn(arr, mid)
+        if val > x:
+            high = mid - 1
+        else:
+            low = mid + 1
+    return high
```

Comment:
Do we have this in Python core library?
https://docs.python.org/3/library/bisect.html

Reply:
Ah, yes, I totally forget about `bisect`. Let me check if I can replace it with `bisect`.

Reply:
Because there is no time, please add TODO comment and do it in the future.

Reply:
Added in https://github.com/openvinotoolkit/datumaro/pull/931/commits/c5730b7fc806bc7de4e486eafeeed0cfb3e726e6